packages/fxa-settings/src/pages/ResetPassword/mocks.tsx

```diff
@@ -31,3 +37,22 @@ export const mockAccountWithUnexpectedError = {
 
 mockAccountWithThrottledError.resetPassword = () =>
   Promise.reject('some error');
+
+export function createMockResetPasswordWebIntegration(): ResetPasswordBaseIntegration {
+  return {
+    type: IntegrationType.Web,
+    getServiceName: jest.fn().mockReturnValue(MozServices.Default),
+  };
+}
+
+export function createMockResetPasswordOAuthIntegration(
+  serviceName = MOCK_SERVICE
+): ResetPasswordOAuthIntegration {
+  return {
```

Comment:
Seems like we do this alot. Much improved, but still a lot of boiler plate. I dug this jest mocking extension up that seems to have decent traction, https://www.npmjs.com/package/jest-mock-extended. Don't really want to add another lib, but that does seem pretty convenient.

Reply:
Oh that does seem interesting. I've already finished a decent amount of this locally, so I think unfortunately that may need to be another ticket if we want to use it. 😩

One thing I was thinking of doing to remove some boilerplate in the meantime is at least sharing the mock/type for those that only need the `type: IntegrationType` reference, probably in `lib/integrations/mocks.tsx`. For example I have this:

```
export interface CompleteResetPasswordBaseIntegration {
  type: IntegrationType;
}

function createMockResetPasswordWebIntegration(): CompleteResetPasswordBaseIntegration {
  return {
    type: IntegrationType.Web,
  };
}

function createMockResetPasswordSyncDesktopIntegration(): CompleteResetPasswordBaseIntegration {
  return {
    type: IntegrationType.SyncDesktop,
  };
}
```

I need this in another test/mock file as well. So for these I could just reference this instead of recreating these for each page (I'm open to suggestions on the interface name, `IIntegrationType` looks sorta weird):

```
export interface IntegrationSubsetType {
  type: IntegrationType;
}

function createMockWebIntegration(): IntegrationSubsetType {
  return {
    type: IntegrationType.Web,
  };
}

function createMockSyncDesktopIntegration(): IntegrationSubsetType {
  return {
    type: IntegrationType.SyncDesktop,
  };
}```

Reply:
@dschom after you take another, let me know if you'd like us to add `jest-mock-extended` and I can file an issue for it. I don't have strong feelings either way but I would like to reuse some of these with storybook where possible. Regardless, I'd like to file an issue to update some function names.